samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -179,20 +189,51 @@ public ApplicationStatus status(StreamApplication streamApp) {
   }
 
   /**
+   * Generates processorId for each {@link StreamProcessor} using the configured {@link ProcessorIdGenerator}
+   *
+   * @param config Application config
+   * @return String that uniquely represents an instance of {@link StreamProcessor} in the current JVM
+   */
+  /* package private */
+  String getProcessorId(Config config) {
```

Comment:
> Shouldn't JobCoordinator be generating the processorId according to SEP-1? 

Yeah. We established that processorId is to be generated by the jobCoordinator. However, if the interface for generating processorID is configurable, then it should ideally come from the user-facing layer. 

> Is this temporary? 

Not sure if it is temporary. There is some blurring of the lines in terms of which component generates the processorId and how it relates to the executing environment. When JobCoordinator was initially designed, Jobcoordinator was taking on the role of coordinator AND as a specification of the executing environment. With ApplicationRunners and the newer interfaces, this assumption seems to have shifted. 

I don't believe there was any deliberate discussion when it was decided that processorId can be moved out of the JobCoordinator layer. Xinyu and I just decided agreed over slack because it was useful for ApplicationRunner to track the lifecycle of the StreamProcessor. However, whether it has to be done via the processorId is a reasonable question. It stems from the fact that there is no clear contract between the applicationrunners and StreamProcessor. My two cents.